language-server/modules/langserver-core/src/test/java/org/ballerinalang/langserver/codeaction/AbstractCodeActionTest.java

```diff
@@ -422,6 +422,32 @@ protected boolean validateAndModifyArguments(JsonObject actualCommand,
                 }
             }
             return false;
+        } else if ("ballerina.action.extract".equals(actualCommand.get("command").getAsString())) {
+            if (actualArgs.size() == 3) {
+                String actualName = actualArgs.get(0).getAsString();
+                String expectedName = expArgs.get(0).getAsString();
+
+                String actualFilePath = actualArgs.get(1).getAsString().replace(sourceRoot.toString(), "");
+                if (actualFilePath.startsWith("/") || actualFilePath.startsWith("\\")) {
+                    actualFilePath = actualFilePath.substring(1);
+                }
+                String expectedFilePath = expArgs.get(1).getAsString();
+                
+                JsonObject actualTextEdits = actualArgs.get(2).getAsJsonObject();
+                JsonObject expectedTextEdits = expArgs.get(2).getAsJsonObject();
+
+                if (actualName.equals(expectedName) && actualFilePath.equals(expectedFilePath)
+                        && actualTextEdits.equals(expectedTextEdits)) {
+                    return true;
+                }
+
+                JsonArray newArgs = new JsonArray();
+                newArgs.add(actualName);
+                newArgs.add(actualFilePath);
+                newArgs.add(actualTextEdits);
+                actualCommand.add("arguments", newArgs);
+            }
+            return false;
```

Comment:
This method can grow very fast if we are to add each command. If this is common for each `extract to constant`, `extract to local var` and `extract to function` we can keep this. If not, shall we override this method in the respective child class?

Reply:
This is common to all three code actions, hence keeping this.